x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java

```diff
@@ -247,7 +247,7 @@ public void testDomainAssignment() throws Exception {
             orders.add(i);
         }
         Collections.shuffle(orders, random());
-        List<String> domains = randomList(1, 3, () -> randomAlphaOfLengthBetween(1, 3));
+        List<String> domains = randomList(1, 3, () -> randomAlphaOfLengthBetween(1, 3)).stream().distinct().collect(Collectors.toList());
```

Comment:
This is fine, given it's a tiny collection in a test, however `distinct()` just puts all the elements into a LinkedHashSet, so it would be cheaper and easier to do
```suggestion
        Collection<String> domains = new HashSet<>(randomList(1, 3, () -> randomAlphaOfLengthBetween(1, 3)));
```

Reply:
I adopted your suggestion. Thanks!